chia/_tests/core/mempool/test_mempool_manager.py

```diff
@@ -1937,3 +1941,71 @@ async def get_coin_records(coin_ids: Collection[bytes32]) -> List[CoinRecord]:
             assert result[1] != MempoolInclusionStatus.FAILED
     except ValidationError as e:
         assert e.code == expected
+
+
+@pytest.mark.anyio
+@pytest.mark.limit_consensus_modes(allowed=[ConsensusMode.HARD_FORK_2_0])
+async def test_fill_rate_block_validation(consensus_mode: ConsensusMode) -> None:
+    """
+    This test creates a puzzle at the maximum cost limit, and generates a block
+    with it, ensuring it passes block validation.
+    """
+
+    async def send_to_mempool(full_node: FullNodeSimulator, spend_bundle: SpendBundle) -> None:
+        res = await full_node.send_transaction(wallet_protocol.SendTransaction(spend_bundle))
+        assert res is not None and ProtocolMessageTypes(res.type) == ProtocolMessageTypes.transaction_ack
+        res_parsed = wallet_protocol.TransactionAck.from_bytes(res.data)
+        assert res_parsed.status == MempoolInclusionStatus.SUCCESS.value
+
+    async def fill_mempool_with_test_sb(full_node_api: FullNodeSimulator) -> Tuple[bytes32, SerializedProgram, bytes32]:
+        puzzle = SerializedProgram.to((1, [[ConditionOpcode.REMARK, b"1" * 12_000]]))
+        ph = puzzle.get_tree_hash()
+        for _ in range(2):
+            await full_node_api.farm_new_transaction_block(FarmNewBlockProtocol(ph))
+        coin_records = await full_node_api.full_node.coin_store.get_coin_records_by_puzzle_hash(False, ph)
+        coin = next(cr.coin for cr in coin_records if cr.coin.amount == 250_000_000_000)
+        coin_spend = make_spend(coin, puzzle, SerializedProgram.to([]))
+        sb = SpendBundle([coin_spend], G2Element())
+        await send_to_mempool(full_node_api, sb)
+        return coin.name(), puzzle, sb.name()
+
+    # This is the cost that our test bundle will have, so we're setting it as
+    # the maximum block cost to ensure a 100% fill rate in this test.
+    TEST_MAX_BLOCK_CLVM_COST = uint64(144_785_330)
+    constants = test_constants.replace(MAX_BLOCK_COST_CLVM=TEST_MAX_BLOCK_CLVM_COST)
+    async with setup_simulators_and_wallets(1, 0, constants) as setup:
+        full_node_api = setup.simulators[0].peer_api
+        assert full_node_api.full_node._mempool_manager is not None
+        # We have to alter these values here as they're not exposed elsewhere
+        # and without them we won't be able to get the test bundle in.
+        full_node_api.full_node._mempool_manager.max_tx_clvm_cost = TEST_MAX_BLOCK_CLVM_COST
+        full_node_api.full_node._mempool_manager.mempool.mempool_info = dataclasses.replace(
+            full_node_api.full_node._mempool_manager.mempool.mempool_info,
+            max_block_clvm_cost=CLVMCost(TEST_MAX_BLOCK_CLVM_COST),
+        )
```

Comment:
is this necessary? what constant does the mempool have if you don't do this?

Reply:
It is, otherwise we get `MAX_BLOCK_COST_CLVM` * `BLOCK_SIZE_LIMIT_FACTOR`, and the latter is not exposed.

Reply:
I see. I think you should explain that in a comment. Once we raise the fill rate to 1, we would want to remove this, right?
Also, why don't you just do:
```
full_node_api.full_node._mempool_manager.max_block_clvm_cost=CLVMCost(TEST_MAX_BLOCK_CLVM_COST)
```
?